### PR TITLE
Allow for seperate admin configuration for trust token

### PR DESCRIPTION
**Pull request by Peterburnett**

### CODE REVIEW

File: factor/token/tests/factor_test.php [hunk 1]
```diff
@@ -69,6 +72,24 @@ public function test_calculate_expiry_time_in_general() {
...
+        $this->setAdminUser();
+
+        // Confirm that the expiry is based on the admin duration.
+        // Duration for regular users is covered by the general calc test.
+        $method = new \ReflectionMethod($this->factor, 'calculate_expiry_time');
+        $method->setAccessible(true);
+
+        $expiry = $method->invoke($this->factor, $timestamp);
+        $this->assertGreaterThan(7 * HOURSECS, $expiry[1]);
+        $this->assertLessThan(9 * HOURSECS, $expiry[1]);
```
Comment (keevan) [inline factor/token/tests/factor_test.php]: Are we able to deconstruct this to something a bit more readable? I'm not certain what value `$expiry[1]` refers to.

File: factor/token/tests/factor_test.php [hunk 2]
```diff
@@ -69,6 +72,24 @@ public function test_calculate_expiry_time_in_general() {
         $this->assertLessThan($resetdelta + DAYSECS + 30, $expiry[1]);
     }
 
+    public function test_calculate_admin_expiry_time() {
```
Comment (keevan) [inline factor/token/tests/factor_test.php:79]: Potentially phpdoc checks might complain here.

File: factor/token/tests/factor_test.php (hunk 1 above)
Comment (Peterburnett) [reply factor/token/tests/factor_test.php]: Updated. The expiry calculator returns a timestamp and the duration, as we need timestamp for the cookie, and one for the secret manager.=

File: factor/token/tests/factor_test.php (hunk 2 above)
Comment (Peterburnett) [reply factor/token/tests/factor_test.php:79]: fixed

**Review by keevan — APPROVED** [review]

**Pull request merged**